src/server/middleware/api/github.js

```diff
@@ -5,61 +5,113 @@ const { isProduction } = require('../../utils')
 
 const cache = new NodeCache({ stdTTL: 900 })
 
-module.exports = async (_, res) => {
+async function getFreshGithubData() {
+  try {
+    const { repository } = await graphql(
+      `
+        {
+          repository(owner: "iterative", name: "dvc") {
+            stargazers {
+              totalCount
+            }
+            issues(
+              first: 3
+              states: OPEN
+              orderBy: { field: CREATED_AT, direction: DESC }
+            ) {
+              edges {
+                node {
+                  title
+                  createdAt
+                  url
+                  comments {
+                    totalCount
+                  }
+                }
+              }
+            }
+          }
+        }
+      `,
+      {
+        headers: {
+          authorization: `token ${process.env.GITHUB_TOKEN}`
+        }
+      }
+    )
+
+    const issues = repository.issues.edges.map(({ node }) => ({
+      title: node.title,
+      url: node.url,
+      comments: node.comments.totalCount,
+      date: node.createdAt
+    }))
+
+    const stars = repository.stargazers.totalCount
+
+    cache.set('issues', issues)
+    cache.set('stars', stars)
+
+    return { issues, stars }
+  } catch (e) {
+    return { error: e.message || e }
+  }
+}
+
+async function issues(_, res) {
   if (!process.env.GITHUB_TOKEN) {
-    res.status(200).json({ issues: [] })
+    res
+      .status(200)
+      .json({ issues: [], error: 'No GITHUB_TOKEN specified on the server!' })
```

Comment:
we don't want to expose implementation details to public

Reply:
Fair, I'll remove it

Reply:
Done! Now it's an empty 403 for both instances.